utils/hct/gen_intrin_main.txt

```diff
@@ -393,6 +393,7 @@ void [[]] RawBufferStore(in u64 addr, in $funcT value);
 void [[]] RawBufferStore(in u64 addr, in $funcT value, in uint alignment);
 void [[]] ext_execution_mode(in uint mode, ...);
 void [[]] ext_execution_mode_id(in uint mode, ...);
+void [[]] ext_literal(in any<> value);
```

Comment:
This also feels odd and unwieldy. Don't we know that the template parameters need to be literals, and that they need to be encoded as immediate values? Why do we need users to specify it this way?

Reply:
Is that not something we should capture as a type requirement rather than an intrinsic? If we use an intrinsic to act as a cast, it seems like it should produce a value with a type annotation rather than returning void where we implicitly "look through" to the argument.

Reply:
We used the intrinsic because it already existed for this purpose.

I'm not sure what you mean by "type requirment", but we could define a new type `vk::Literal` and then cast the HLSL literal to that type. Is that what you are thinking about?